src/Features/Core/Portable/Completion/Providers/SymbolCompletionItem.cs

```diff
@@ -214,22 +216,28 @@ public static SupportedPlatformData GetSupportedPlatforms(CompletionItem item, W
             return null;
         }
 
-        public static int GetContextPosition(CompletionItem item)
+        public static async Task<int> GetContextPositionAsync(
+            Document document, CompletionItem item, CancellationToken cancellationToken)
         {
-            if (item.Properties.TryGetValue("ContextPosition", out var text) && int.TryParse(text, out var number))
+            if (item.Properties.TryGetValue("ContextPosition", out var text) &&
+                int.TryParse(text, out var number))
             {
-                return number;
+                // We have no access to the editor at this layer.  So it's not 
```

Comment:
How did we find ourselves in this situation in the first place? Doesn't the rest of the completion system do the mapping before it gets into the providers to ensure this sort of a problem doesn't happen? This seems like a very leaky abstraction.